tests/20-SideMenu/SideMenu.test.tsx

```diff
@@ -24,37 +34,131 @@ const { SideMenuPreview } = composeStories(stories);
 
-  //TODO: fill this
+  const sections = document.getElementsByTagName("section");
+  expect(sections).toHaveLength(7);
+
+  expect(within(sections[2]).getByText("Recent boards")).toBeVisible();
+  expect(within(sections[2]).getAllByRole("link")).toHaveLength(4);
+
+  expect(within(sections[3]).getByText("Fandom boards")).toBeVisible();
+  expect(within(sections[3]).getAllByRole("link")).toHaveLength(3);
+
+  expect(within(sections[4]).getByText(Regular!.args!.title!)).toBeVisible();
+  expect(within(sections[4]).getAllByRole("link")).toHaveLength(
+    (Regular.args as BoardsSectionProps).boards!.length
+  );
+
+  expect(within(sections[5]).getByText(Loading!.args!.title!)).toBeVisible();
+  expect(
+    within(sections[5]).getAllByLabelText("board is loading")
+  ).toHaveLength((Loading.args as LoadingSectionProps).placeholdersCount);
+
+  expect(within(sections[6]).getByText(Empty!.args!.title!)).toBeVisible();
+  expect(within(sections[6]).queryAllByRole("link")).toHaveLength(0);
+  expect(
+    within(sections[6]).getByText((Empty.args as EmptySectionProps).emptyTitle)
+  ).toBeVisible();
+  expect(
+    within(sections[6]).getByText(
+      (Empty.args as EmptySectionProps).emptyDescription
+    )
+  ).toBeVisible();
 });
 
-//Currently not rendered in Storybook. Need to add menuOptions arg in SideMenu story before testing
 test("Renders boards dropdown", async () => {
   render(<SideMenuPreview />);
 
-  //TODO: fill this
+  const actionReturn = jest.fn();
+  mocked(action).mockReturnValue(actionReturn);
+
+  userEvent.click(screen.getByLabelText("board menu options"));
+  await waitFor(() => {
+    expect(screen.getByText("Dismiss notifications")).toBeVisible();
+  });
+  userEvent.click(screen.getByText("Dismiss notifications"));
+  await waitFor(() => {
+    expect(action).toBeCalledWith("dismissNotifications");
+  });
+
+  await waitFor(() => {
+    expect(screen.queryByText("Dismiss notifications")).not.toBeInTheDocument();
+  });
 });
 
 test("Renders board filter", async () => {
   render(<SideMenuPreview />);
 
-  //TODO: fill this
+  expect(screen.getByRole("searchbox")).toHaveAccessibleName("board filter");
+  expect(screen.getByRole("searchbox")).toEqual(
+    screen.getByPlaceholderText("Filter boards")
+  );
 });
 
 test("Correctly propagates filter change on text entry", async () => {
   render(<SideMenuPreview />);
 
-  //TODO: fill this
+  const actionReturn = jest.fn();
+  mocked(action).mockReturnValue(actionReturn);
+
+  const boardFilter = screen.getByRole("searchbox");
+  userEvent.type(boardFilter, "meta");
+  await waitFor(() => {
+    expect(action).toBeCalledWith("filterBoards");
+    expect(actionReturn).toBeCalledWith("meta");
+  });
+  expect(boardFilter).toHaveValue("meta");
+
+  userEvent.type(boardFilter, "{backspace}{backspace}");
+  await waitFor(() => {
+    expect(action).toBeCalledWith("filterBoards");
+    expect(actionReturn).toBeCalledWith("me");
+  });
+  expect(boardFilter).toHaveValue("me");
```

Comment:
Can you add a similar case, but rather than just checking "me" it deletes the whole word and checks that it's still called with empty string?

I would say to make it its own case.

Reply:
Done